extensions/ql-vscode/src/extension.ts

```diff
@@ -216,16 +226,22 @@ export async function activate(ctx: ExtensionContext): Promise<void> {
 
   ctx.subscriptions.push(distributionConfigListener.onDidChangeDistributionConfiguration(() => installOrUpdateThenTryActivate({
     isUserInitiated: true,
-    shouldDisplayMessageWhenNoUpdates: false
+    shouldDisplayMessageWhenNoUpdates: false,
+
+    // only auto update on startup if the user has previously requested an update
+    // otherwise, ask user to accept the update
+    avoidAutoUpdating: !!ctx.globalState.get(shouldUpdateOnNextActivationKey)
```

Comment:
This call to `installOrUpdateThenTryActivate` happens when a user changes the distribution configuration, for example by changing the CodeQL executable path setting.  I think this logic for `shouldUpdateOnNextActivationKey` belongs in the final call to `installOrUpdateThenTryActivate` on lines 241 to 245.

Reply:
Darn. I messed that up. Fixed now. And a bit of refactoring/renaming that I hope makes things clearer.